mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```
@@ -443,46 +516,22 @@ struct TransposeRewritePattern : public OpRewritePattern<tosa::TransposeOp> {
       top->replaceAllUsesWith(convOp);
     } else if (tosa::MatMulOp matMulOp =
                    tInput.getDefiningOp<tosa::MatMulOp>()) {
-      setTranspose(matMulOp, "transpose_c", matmulNonTrivial);
+      LogicalResult res = checkMatMulTransposeValid(matMulOp, dims);
+      if (res.failed()) {
```

You can do `if(failed(checkMatMulTranspose(...)) { return failure(); }` which is somewhat more the style.

---

Or `.failed() { return failure(); }`

---

I want to propagate the error.

---

There isn't an error. `LogicalResult` is a wrapper around `bool`

---

right.. need to learn more about diagnostic handling. done.